kubernetes/metadata/resource.go

```diff
@@ -110,7 +111,11 @@ func (r *Resource) GenerateK8s(kind string, obj kubernetes.Resource, options ...
 	}
 
 	// Exclude any labels that are present in the exclude_labels config
-	for _, label := range r.config.ExcludeLabels {
+	var labelMaptoExclude mapstr.M
+	if len(r.config.ExcludeLabels) != 0 {
+		labelMaptoExclude = generateMapSubset(accessor.GetLabels(), r.config.ExcludeLabels, r.config.LabelsDedot)
```

Comment:
I'm trying to understand what could be the implication of changing this 🤔 . Does any of the existent functionality changes?

Reply:
Excluding with regex can lead to situations where we exclude more things than we want. So might be prone to logical errors for users. That is why I am a little sceptical to introduce it and not leave the previous functionality to delete them specifically